raiden_contracts/deploy/__main__.py

```diff
@@ -729,6 +746,28 @@ def verify_deployed_service_contracts(
         f'matches the compiled data from contracts.json',
     )
 
+    user_deposit = verify_deployed_contract(
+        web3,
+        contract_manager,
+        deployment_data,
+        CONTRACT_USER_DEPOSIT,
+    )
+
+    # We need to also check the constructor parameters against the chain
+    constructor_arguments = deployment_data['contracts'][
+        CONTRACT_USER_DEPOSIT
+    ]['constructor_arguments']
+    assert len(constructor_arguments) == 1
+    assert to_checksum_address(
+        user_deposit.functions.token().call(),
+    ) == token_address
+    assert token_address == constructor_arguments[0]
```

Comment:
Isn't it somewhat redundant to do both
```python
assert to_checksum_address(
    user_deposit.functions.token().call(),
) == token_address
```
and
```python
constructor_arguments = deployment_data['contracts'][
    CONTRACT_USER_DEPOSIT
]['constructor_arguments']
assert len(constructor_arguments) == 1
assert token_address == constructor_arguments[0]
```

Reply:
The first checks what's onchain.  The second checks what we remember from the deployment time.

So the second check might be redundant, but I'm not sure. `deployment_data` sometimes comes from the filesystem.